soh/soh/Enhancements/randomizer/randomizer_grotto.c

```diff
@@ -143,7 +143,8 @@ s16 Grotto_OverrideSpecialEntrance(s16 nextEntranceIndex) {
     // If Link hits a grotto exit, load the entrance index from the grotto exit list
     // based on the current grotto ID
     if (nextEntranceIndex == 0x7FFF) {
-        // SaveFile_SetEntranceDiscovered(0x0800 + grottoId);
+        Entrance_SetEntranceDiscovered(0x0800 + grottoId);
+        EntranceTracker_SetLastEntranceOverride(0x0800 + grottoId);
```

Comment:
couple more grotto `0x800` magic numbers, might be good to `#define` it somewhere for clarity